app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarView.kt

```diff
@@ -227,7 +227,23 @@ class BrowserToolbarView(
 
     fun expand() {
         if (!settings.shouldUseBottomToolbar) {
-            layout.app_bar?.setExpanded(true)
+            view.app_bar?.setExpanded(true)
+        }
+    }
+
+    /**
+     * Performs all actions that refer to the BrowserToolbarView that don't require user action.
+     * They are set in the callback because we need to ensure phones execute the code before
+     * we are done inflating.
+     */
+    fun loadAsync(action: View.() -> Unit) {
```

Comment:
My overall suggestion is to change this to `(View) -> Unit` and, instead of accessing `browserToolbarView.view` in the caller, access the argument passed in instead. There are several things going on here:
- `lateinit var browserToolbarView.view` is not type-safe: if we pass the view in instead, we can guarantee its type safety. In general, passing arguments directly into functions, instead of accessing them through members, is safer (e.g. can't be rebound, types are as up-to-date as possible, etc.) and is one of the primary reasons functional programming is gaining popularity
- Adding a receiver (i.e. `View.`) should be used carefully since it binds `this` in the caller. This is an issue because 1) it's often not clear which `this` is being referred to and 2) `this` can be used without explicitly writing `this` which becomes a common source of bugs, especially when refactoring. In general, receivers should only used for very small blocks of code or in very specific contexts where the meaning of `this` is well known. How `this` is used in JavaScript is a good example of why this is an unsafe pattern.